File: python/main/network/mqtt.py
import paho.mqtt.client as mqtt_client

import json
import random

broker = "10.0.0.205"
port = 1883
client_id = f'python-mqtt-{random.randint(0, 1000)}'
username = 'ben'
password = 'imarobot'
client = mqtt_client.Client(client_id)

def publish(topic, payload):
    try:
        v = json.dumps(payload)

        result = client.publish(topic, v)
        # result: [0, 1]
        status = result[0]
        if status == 1:
            print(f"Failed to send message to topic {topic}")
    except (ConnectionResetError, ConnectionRefusedError) as err:
            print(err)

def connect(on_connect, on_message, on_disconnect):

    client.username_pw_set(username, password)
    client.on_connect = on_connect
    client.connect(broker, port, 30)
    client.on_message = on_message
    client.on_disconnect = on_disconnect
    client.loop_start()

File: transcribe_service/transcribe_service/src/main/kotlin/scorpion/mqtt/MQTT.kt
package mqtt

import kotlinx.coroutines.delay
import org.eclipse.paho.client.mqttv3.MqttClient
import org.eclipse.paho.client.mqttv3.MqttConnectOptions
import org.eclipse.paho.client.mqttv3.MqttMessage
import org.eclipse.paho.client.mqttv3.persist.MemoryPersistence
import scorpion.mqtt.Command
import scorpion.mqtt.Topic
import java.util.*

class MQTT(private val listener: MqttListener, private val broker: String) {



    private val clientId = UUID.randomUUID().toString()
    private val persistence = MemoryPersistence()
    private lateinit var client : MqttClient



     suspend fun start() {
         client = MqttClient(broker, clientId, persistence)


            val connOpts = MqttConnectOptions()
            connOpts.isCleanSession = true
            connOpts.userName = "ben"
            connOpts.password = "imarobot".toCharArray()
            println("MQTT Connecting to broker: $broker")
            client.connect(connOpts)


            client.setCallback(listener)
            while (client.isConnected.not()) {
                delay(10)
            }
           listener.onConnected()

    }

     fun publish(topic: Topic, payload: String) {


        val message = MqttMessage(payload.toByteArray())
        message.qos = qos
         if (client.isConnected) {
             client.publish(topic.name, message)
         }


    }

     fun subscribe(topic: String) {
       // println("MQTT SUB $topic")
        client.subscribe(topic)


    }

    fun stop() {
        client.disconnect()
    }


    companion object {
        private const val qos = 0
    }

}